### PR TITLE
Installation page

### PR DESCRIPTION
**Pull request by treba123**
**@ Codacy fail:**
Codacy is failing on this PR as libs like jquery get reviewed, too, which do not pass the test. The code contributions from me are fine.

This is a working version of the installation page https://github.com/loklak/loklak_server/issues/519
Warning: requires bash for shell interaction!

It currently offers:
- admin account creation
- a user registration setting
- general settings like host- and shortlink url, peername, backend settings
- smtp settings (with test option)
- https settings

The installation is started with an own start script, in it's own main class. Launching it will be offered by the normal start script.
In the process I also made some changes/additions to the email handler

How to test:
- clonegit@github.com:treba123/loklak_server.git
- checkout development
- build and start via start.sh
- choose to do the installation
- go through the dialog
- afterwards, the normal server should be started

**Codacy:**
I adopted most recommendations, most remaining issues are with external files which I have little influence on.

### CODE REVIEW

**Comment by mariobehling** [comment]
> 3. what happens on already deployed servers that just get updated? (currently the installation gets disabled in the customized_config.properties, but we can't set this value on upgrade)
4. should the installation provide a restart button (needed for some settings to take effect) or should the user do that manually?

As loklak is constantly providing updates/upgrades the goal was always to provide upgrades that work without the need of any user actions. Do I understand you correctly, that this is no longer possible in the way you propose it here?

**Comment by mariobehling** [comment]
Please disregard my previous comment and let's move this discussion to the issue itself #519

**Comment by daminisatya** [comment]
@treba123 is this out for review?

**Comment by treba123** [comment]
@daminisatya yep :) was just about to ask on gitter.

File: src/org/loklak/LoklakInstallation.java [hunk 1]
```diff
@@ -0,0 +1,478 @@
+/**
+ *  LoklakServer
+ *  Copyright 22.02.2015 by Michael Peter Christen, @0rb1t3r
```
Comment (daminisatya) [inline src/org/loklak/LoklakInstallation.java]: @treba123  I guess you need to change this line.

**Comment by shivenmian** [comment]
@treba123 Please squash before you merge :)

**Comment by treba123** [comment]
Will do :)

**Comment by treba123** [comment]
Hm somehow the commit fails to build, but runs fine on my mashine. Any ideas @all ?

**Comment by treba123** [comment]
@sudheesh001 this pr also includes some changes to the login/signup pages as I had to make some changes on the backend and therefor went ahead to also make some small changes on the frontend aswell. 
They don't change anything fundamentally but make use of native browser functions, which mostly benefits touch devices and reduce code complexity a bit.
Hope it's alright.

**Comment by sudheesh001** [comment]
screen shot 2016-08-11 at 10 41 39 am

Button breaks and are not clickable, keeping the home page at the installation wizard

**Comment by sudheesh001** [comment]
Hitting (r) Remind me next time also takes me to the installation wizard page,it should instead take me to the home page.

**Comment by treba123** [comment]
@sudheesh001 the password is not not sufficient. I'll make the hint for that more prominent.
About hitting r, what's the output of the start script? Does it report 'starting loklak' or 'starting loklak installation' ?
Because I noticed that quite often if you first started the installation, the browser does not refresh afterwards. Could you check this by emtying the browser cache or using a second browser?

Thanks for testing!

**Comment by treba123** [comment]
@sudheesh001 I think I found a solution for the browser caching issue, so the issues you faced should be solved. Could you retest at some point?

**Comment by treba123** [comment]
Relabel to 'requesting review' as the issues encountered should be fixed.

**Comment by jig08** [comment]
@treba123 `codacy` checks fail, kindly review

**Comment by treba123** [comment]
@jig08 this is not because of my code but of other libs we're using (like jquery). See first post.
They should not get reviewed.

**Comment by sudheesh001** [comment]
Reviewing.

**Comment by sudheesh001** [comment]
screen shot 2016-08-17 at 4 39 46 pm
Same issue with the installation page.
What is going well
[x] Hitting `n` takes to loklak home page
[x] Remind me next time seems to be working

What is not going well
[ ] Hitting the continue button after filling the form items
[ ] There should be a UI Button to also `Skip Installation` from the Web UI.

**Comment by treba123** [comment]
@sudheesh001 well, there is already the abort button, which ends the installation. I don't see why there would be a need for a skip button there, as to start the installation you have to explicitly type 'y' in the script. In what case would that be of benefit?

At the password thing: are you really sure your password meets the requirements as written in the hint? Because I tested it on chrome here and it works fine...

**Comment by sudheesh001** [comment]
Abort is generally used in places where you'd like to Stop the complete execution, Skip refers to skip a step. Generally speaking the Abort should stop the loklak server in the installation, similar to how hitting Abort/Exit should stop any installation process that's going on. So the Abort and Skip are two different functionalities altogether. I hope it's more clear now

**Comment by treba123** [comment]
Ah alright. But then 'skip installation' does not make sense here (IMO), as the installation is a whole process in itself. So the 'abort' refers to the installation, which can also be launched on it own and is not necessarily part of the normal start procedure.

**Comment by sudheesh001** [comment]
The functionality looks okay but I really think we need a `Skip Install for now / Remind me later` button on the UI itself. I think @Orbiter should make the review.

**Comment by Orbiter** [comment]
Hi, that is good work, however there are some things which did not work when I tested this:

- startup of bin/installation.sh and entering of the installation detailes worked smoothly. Very good work! One button on the first page looks a bit shifted to the side, but whatever
- after the shutdown, I started normally but loklak complained that the installation was not done:
```
$:loklak_server admin$ bin/installation.sh
starting loklak installation
loklak installation started at port 9000, open your browser at http://localhost:9000
$:loklak_server admin$ bin/start.sh
Loklak detected that you did not yet run the installation wizard.
It let's you setup an administrator account and a number of settings, but is not mandatory.
You can manually start it by running bin/installation.sh
Would you like to start the installation now? (y)es, (n)o, (r)emind me next time
```
However, while this message was shown, the server started up nevertheless.

I also checked the customized_config.properties and found some problems:
```
backend=http\://loklak.org
host.url=http\://127.0.0.1\:9000
shortlink.urlstub=http\://127.0.0.1\:9000
```
These urls look broken.
Only the last thing looks like a bug, the other things are more or less cosmetics. Can you please have a look for the strange urls?

**Comment by treba123** [comment]
@Orbiter thanks for the review!
The urls are actually right, I wondered about this, too. Here is an [explanation](https://docs.oracle.com/javase/7/docs/api/java/util/Properties.html#store(java.io.Writer,%20java.lang.String)) 'The key and element characters #, !, =, and : are written with a preceding backslash to ensure that they are properly loaded.'
When loaded by the Loklak main class, the urls a fine again.

About the installation reminder: right, I missed that. Will fix :)

**Comment by Orbiter** [comment]
Strange thing with the properties encoding but ok.

About the recently discussed abort option: I don't think we need that on the web page. But we would require some transparency about the whole process when the installation wizard starts, like:
- print on the terminal that the server terminates after the wizard was run successfully
- have the option to stop the wizard with bin/stop.sh (I would guess that will work anyway?)

**Pull request merged**